src/udev/cryptsetup_id/cryptsetup_id.c

```diff
@@ -0,0 +1,148 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+
+#include <getopt.h>
+#include <libcryptsetup.h>
+
+#include "build.h"
+#include "cryptsetup-util.h"
+#include "device-private.h"
+#include "device-util.h"
+#include "main-func.h"
+#include "udev-util.h"
+
+static const char *arg_device = NULL;
+
+static int help(void) {
+        printf("%s [OPTIONS...] [PATH]\n\n"
+               "  -h --help     Show this help text\n"
+               "     --version  Show package version\n",
+               program_invocation_short_name);
+        return 0;
+}
+
+static int parse_argv(int argc, char *argv[]) {
+        static const struct option options[] = {
+                { "help",     no_argument, NULL, 'h' },
+                { "version",  no_argument, NULL, 'v' },
+                {}
+        };
+        int c;
+
+        while ((c = getopt_long(argc, argv, "h", options, NULL)) >= 0)
+                switch (c) {
+                case 'h':
+                        return help();
+                case 'v':
+                        return version();
+                case '?':
+                        return -EINVAL;
+                default:
+                        assert_not_reached();
+                }
+
+        if (argc > 2)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Error: unexpected argument.");
+
+        arg_device = argv[optind];
+        return 1;
+}
+
+
+static int print_attribute(const char* name, const char* devname) {
+        _cleanup_(sd_device_unrefp) struct sd_device *dev = NULL;
+        char const *syspath, *sysname;
+        int r;
+
+        r = sd_device_new_from_devname(&dev, devname);
+        if (r < 0)
+                return log_debug_errno(r, "Could not resolve device '%s': %m", devname);
+
+        r = sd_device_get_syspath(dev, &syspath);
+        if (r < 0)
+                return log_debug_errno(r, "Could not resolve syspath for device '%s': %m", devname);
+
+        r = sd_device_get_sysname(dev, &sysname);
+        if (r < 0)
+                return log_debug_errno(r, "Could not resolve sysname for device '%s': %m", devname);
+
+        printf("CRYPT_%s=%s\n", name, syspath);
+        printf("CRYPT_%s_SYSNAME=%s\n", name, sysname);
+        return 0;
+}
+
+static int run(int argc, char *argv[]) {
+        _cleanup_(sd_device_unrefp) struct sd_device *dev = NULL;
+        _cleanup_(crypt_freep) struct crypt_device *cd = NULL;
+        const char *devnode, *type, *device_name, *metadata_device_name;
```

Comment:
Please move declarations of `device_name` and `metadata_device_name` to the `else` block below.